Adapters/Vungle/ISVungleAdapter/ISVungleRewardedVideoAdapterRouter.m

```diff
@@ -0,0 +1,125 @@
+//
+//  ISVungleRewardedVideoAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleRewardedVideoAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleRewardedVideoAdapterRouter()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleRewardedVideoAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                      parentAdapter:(ISVungleAdapter *)parentAdapter
+                           delegate:(id<ISRewardedVideoAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _parentAdapter = parentAdapter;
+        _delegate = delegate;
+        _bidPayload = nil;
+        _isNeededInitCallback = NO;
+    }
+
+    return self;
+}
+
+- (void)loadRewardedVideoAd {
+    self.rewardedVideoAd = [[VungleRewarded alloc] initWithPlacementId:self.placementID];
+    self.rewardedVideoAd.delegate = self;
+
+    // Because there's no auto-caching in 7.0 VungleAds, the Ad would
+    // never be ready until a load call is made. We don't need to
+    // check [VungleRewarded canPlayAd] here.
+    [self.rewardedVideoAd load:self.bidPayload];
```

Comment:
u should Move line 34 and 35 into initWithPlacementID: method. That way we create the object at init level. 

And u should wrap this line 40 in `if(!CanPlay){}`
U should do the same for all ad Formates. 
Reason if the load ad gets called after the SDK already loaded ad. SDK returns error. If u have this check here u dont have to call SDK to load but u can just just Do Line 66&73 in the else block. which covers a lot of cases.

Reply:
But with the current implementation if pubs called loadRewardedVideoForBiddingWithAdapterConfig back 2 back we are creating multiple instances here. 
So i think we should change to my suggestion. @YueVungle

Reply:
now that i have relooked the code. hold this coment. i raised a qn to IS will do something based on the reply.